File: src/simulator/core/src/com/kristoff/robomaster_simulator/envs/launchers/py_entrypoints/EntryPoint.java
package com.kristoff.robomaster_simulator.envs.launchers.py_entrypoints;

import com.kristoff.robomaster_simulator.envs.environment.Environment;
import com.kristoff.robomaster_simulator.envs.launchers.LauncherBase;
import py4j.GatewayServer;

public class EntryPoint extends LauncherBase {

    public static void main(String[] args) {
        GatewayServer gatewayServer = new GatewayServer(new EntryPoint());
        gatewayServer.start();
        System.out.println("Gateway Server Started");
    }

    @Override
    public void init(){
        super.init();
    }

    @Override
    public Environment getEnv(){
        return env;
    }
}

File: src/env/rllib/entrypoint.py
from py4j.java_gateway import JavaGateway

class entrypoint:
    def __init__(self):
        self.gateway = JavaGateway()

    def seed(self):
        self.env.seed()
    
    def reset():
        self.env.reset

ep = entrypoint()
ep.gateway.entry_point.setA("hello world")
print(ep.gateway.entry_point.getA())

